smoketests/tests/__helpers/npm.ts

```diff
@@ -1,17 +1,50 @@
-import { SpawnOptionsWithoutStdio } from 'child_process';
-import { asyncSpawn } from './asyncSpawn';
+import fs from 'fs/promises';
+import path from 'path';
+import { asyncSpawn, AsyncSpawnOptions, AsyncSpawnResult } from './asyncSpawn.js';
 
-export function npm(cwd: string, args: string[], options?: SpawnOptionsWithoutStdio) {
+export function npm(cwd: string, args: string[], options?: AsyncSpawnOptions) {
```

Comment:
we talked about this on Slack but I still dont get it:
Questions:
- Why do we need to write scripts around this. Why we cannot just use CICD pipeline setup.
- if we're already in the npm workspace, why the development process is hard? We can just have a watch process. We don't need to install every time npm packages or clean them. 
- Why it's not enough to use npm workspace and the same actions. We can execute all actions in a single workspace that has a task node:e2e or browser:e2e. 
- Why do we need to have a dedicated project for smoke tests?

Reply:
>Why do we need to write scripts around this. Why we cannot just use CICD pipeline setup.
We want the developer to be able to run the tests without CICD pipelines.

>if we're already in the npm workspace, why the development process is hard? We can just have a watch process. We don't need to install every time npm packages or clean them.
You can definitely do that if you want. Install NPM packages yourself, set up a watch process, and run the test.

>Why it's not enough to use npm workspace and the same actions. We can execute all actions in a single workspace that has a task node:e2e or browser:e2e.
Installing the packages by itself is a test. I wanted to separate this.

>Why do we need to have a dedicated project for smoke tests?
Why not? Where you would put it otherwise?